### ISSUE TITLE
About _apply_non_overlapping_constraints

### ISSUE THREAD

**Issue by King-king424** [facebookresearch/sam2#527]
I don't quite understand exactly what this function does. Why are these constraints on the prediction mask logits under the batch dimension when batch_size >1.
Hopefully someone can explain the following examples of what it is used for. Thank you very much.
